components/atom/backToTop/src/calcBackToTopEngine.js

```diff
@@ -0,0 +1,41 @@
+const getTarget = value => {
+  if (value instanceof HTMLElement) {
+    return value
+  } else if (value === window.document) {
+    return value.body
+  } else if (value instanceof Object) {
+    return value.current
+  } else {
+    return value
+  }
+}
+
+export const isDocumentElement = target => target === window.document
+
+const calcBackToTopEngine = (target = window.document, minHeight) => {
+  let properties = {}
+  const element = getTarget(target)
+  const isDocument = isDocumentElement(target)
+  const scrollTop = isDocument ? window.scrollY : element.scrollTop
+  const clientHeight = isDocument
+    ? window.innerHeight - (element.offsetHeight - element.clientHeight)
+    : element.clientHeight
+  const offsetHeight = isDocument ? window.innerHeight : element.offsetHeight
+  const scrollHeight = isDocument ? element.scrollHeight : element.scrollHeight
+  const borderHeight = offsetHeight - clientHeight
+  properties = {
+    ...properties,
```

Comment:
Variable `properties` is spreaded and reassigned but there are no other properties inside. Why not create it directly? @andresin87

Reply:
✅